stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -0,0 +1,634 @@
+    try
+        ARTIFACTS_DIR_OVERRIDE[] = artifacts_dir
+        f()
+    finally
+        ARTIFACTS_DIR_OVERRIDE[] = nothing
+    end
+end
+
+"""
+    artifacts_dirs(args...)
+
+Return a list of paths joined into all possible artifacts directories, as dictated by the
+current set of depot paths and the current artifact directory override via the method
+`with_artifacts_dir()`.
+"""
+function artifacts_dirs(args...)
+    if ARTIFACTS_DIR_OVERRIDE[] === nothing
+        return [abspath(depot, "artifacts", args...) for depot in Base.DEPOT_PATH]
+    else
+        # If we've been given an override, use _only_ that directory.
+        return [abspath(ARTIFACTS_DIR_OVERRIDE[], args...)]
+    end
+end
+
+"""
+    ARTIFACT_OVERRIDES
+
+Artifact locations can be overridden by writing `Override.toml` files within the artifact
+directories of Pkg depots.  For example, in the default depot `~/.julia`, one may create
+a `~/.julia/artifacts/Override.toml` file with the following contents:
+
+    78f35e74ff113f02274ce60dab6e92b4546ef806 = "/path/to/replacement"
+    c76f8cda85f83a06d17de6c57aabf9e294eb2537 = "fb886e813a4aed4147d5979fcdf27457d20aa35d"
+
+    [d57dbccd-ca19-4d82-b9b8-9d660942965b]
+    c_simple = "/path/to/c_simple_dir"
+    libfoo = "fb886e813a4aed4147d5979fcdf27457d20aa35d""
+
+This file defines four overrides; two which override specific artifacts identified
+through their content hashes, two which override artifacts based on their bound names
+within a particular package's UUID.  In both cases, there are two different targets of
+the override: overriding to an on-disk location through an absolutet path, and
+overriding to another artifact by its content-hash.
+"""
+const ARTIFACT_OVERRIDES = Ref{Union{Dict{Symbol,Any},Nothing}}(nothing)
+function load_overrides(;force::Bool = false)
+    if ARTIFACT_OVERRIDES[] !== nothing && !force
+        return ARTIFACT_OVERRIDES[]
+    end
+
+    # We organize our artifact location overrides into two camps:
+    #  - overrides per UUID with artifact names mapped to a new location
+    #  - overrides per hash, mapped to a new location.
+    #
+    # Overrides per UUID/bound name are intercepted upon Artifacts.toml load, and new
+    # entries within the "hash" overrides are generated on-the-fly.  Thus, all redirects
+    # mechanisticly happen through the "hash" overrides.
+    overrides = Dict{Symbol,Any}(
+        # Overrides by UUID
+        :UUID => Dict{Base.UUID,Dict{String,Union{String,SHA1}}}(),
+
+        # Overrides by hash
+        :hash => Dict{SHA1,Union{String,SHA1}}(),
+    )
+
+    for override_file in reverse(artifacts_dirs("Overrides.toml"))
+        !isfile(override_file) && continue
+
+        # Load the toml file
+        depot_override_dict = parse_toml(override_file)
+
+        function parse_mapping(mapping::String, name::String)
+            if !isabspath(mapping) && !isempty(mapping)
+                try
+                    mapping = Base.SHA1(mapping)
```

Comment:
there's no `tryparse` for `Base.SHA1`? Perhaps should be added?

Reply:
Agreed.